src/main/java/bio/terra/cli/service/WorkspaceManager.java

```diff
@@ -228,89 +245,224 @@ public RoleBindingList listUsersOfWorkspace() {
+      @Nullable String location) {
+    return createResource(
+        name,
+        () ->
+            new WorkspaceManagerService(
+                    globalContext.server, globalContext.requireCurrentTerraUser())
+                .createControlledGcsBucket(
+                    workspaceContext.getWorkspaceId(),
+                    name,
+                    description,
+                    cloningInstructions,
+                    accessScope,
+                    gcsBucketName,
+                    defaultStorageClass,
+                    lifecycleRules,
+                    location));
+  }
+
+  /**
+   * Add a Big Query dataset as a controlled resource in the workspace. Also updates the cached list
+   * of resources.
+   *
+   * @param name name of the resource. this is unique across all resources in the workspace
+   * @param description description of the resource
+   * @param cloningInstructions instructions for how to handle the resource when cloning the
+   *     workspace
+   * @param accessScope access to allow other workspaces users
+   * @param bigQueryDatasetId Big Query dataset id
+   *     (https://cloud.google.com/bigquery/docs/datasets#dataset-naming)
+   * @param location Big Query dataset location (https://cloud.google.com/bigquery/docs/locations)
+   * @return the resource description object that was created
+   */
+  public ResourceDescription createControlledBigQueryDataset(
+      String name,
+      String description,
+      CloningInstructionsEnum cloningInstructions,
+      AccessScope accessScope,
+      String bigQueryDatasetId,
+      @Nullable String location) {
+    return createResource(
+        name,
+        () ->
+            new WorkspaceManagerService(
+                    globalContext.server, globalContext.requireCurrentTerraUser())
+                .createControlledBigQueryDataset(
+                    workspaceContext.getWorkspaceId(),
+                    name,
+                    description,
+                    cloningInstructions,
+                    accessScope,
+                    bigQueryDatasetId,
+                    location));
+  }
+
+  /**
+   * Create a new resource in the workspace. Also updates the cached list of resources.
    *
-   * @param resourceType type of resource to create
    * @param resourceName name of resource to create
-   * @return the cloud resource that was created
+   * @param createResource function pointer to execute the create request for a particular
+   *     stewardship + resource type combination
+   * @param <T> type of the create response object. this is usually very similar to the
+   *     ResourceDescription object, but not quite identical.
+   * @return the resource description object that was created
    */
-  public CloudResource createControlledResource(
-      CloudResource.Type resourceType, String resourceName) {
+  private <T> ResourceDescription createResource(String resourceName, Supplier<T> createResource) {
```

Comment:
Do we need to take a Supplier as an argument and control when it's called, or is it ok to just take a `T resource` and rename this. Or maybe this isn't doing enough at all? Not totally against, but not totally sold on this function.

Reply:
If we keep this function, then it needs to take a `Supplier` because we need to do the name validation check before creating the resource.

For both `createResource` and `deleteResource`, they just make sure all the create/delete methods update the cache correctly. I went back and forth over whether this is worth it or not. Happy to remove -- it just means we have the same couple lines before and same couple lines after in all create/delete functions.

Reply:
I prefer having business logic validation in the main CLI code, instead of in the command classes. The only validation I think makes sense in the command classes is argument missing/bad argument combination type validation. Maybe a moot point here though, since this logic will likely be removed (either completely or pushed into WSM).